File: leetcode/roman-to-integer.js
/**
 * @param {string} s
 * @return {number}
 */

 //https://leetcode.com/problems/roman-to-integer/description/
 
var romanToInt = function(s) {
    
    var splitted = s.split("");
    
    var sum = 0;
    for(var x =0; x < splitted.length; ++x ){
        var twoDigitsFlag = false;
        if(splitted[x+1]!=null){
            var twoDigits =  splitted[x] + splitted[x+1];    
            twoDigitsFlag = true;
        }
        else{
            var twoDigits = splitted[x];
        }
        
        
        if( x<splitted.length && (["IX","IV","XL","XC","CD","CM"].indexOf(twoDigits)!=-1) && twoDigitsFlag==true ){
         
            switch(twoDigits){
                case 'IV':
                    sum += 4;
                break;
                case 'IX':
                    sum += 9;
                break;
                case 'XL':
                    sum += 40;
                break;
                case 'XC':
                    sum += 90;
                break;
                case 'CD':
                    sum += 400;
                break;
                case 'CM':
                    sum += 900;
                break;
            }
            ++x;
            continue;
         }
        else{
           
            
         switch(splitted[x]){
                
            case 'I':
                sum += 1;
               
                break;
            case 'V':
                sum += 5;
                break;
            case 'X':
                sum += 10;
                break;
            case 'L':
                sum += 50;
                break;
            case 'C':
                sum += 100;
                break;
            case 'D':
                sum += 500;
                break;
            case 'M':
                sum += 1000;
                break;
                
                
                
        }
       
    }

    }
    return sum;
        

    
};

File: leetcode/roman-to-integer.py
#https://leetcode.com/problems/roman-to-integer/
class Solution:
    def romanToInt(self, s: str) -> int:
        
        
        newStack = []
        basicDict ={"M":1000,"D":500,"C":100,"L":50,"X":10,"V":5,"I":1,"CD":400,"CM":900,"XC":90,"XL":40,"IV":4,"IX":9}
        for i,x in enumerate(s):
            if(i>0):
                if(x in ["M","D","C","L","X","V"]):
                    lastItem = newStack[-1]
                    if(lastItem in ["C","X","I"]):
                        if((lastItem+x) in basicDict.keys()):
                            lastItem = newStack.pop()
                            newStack.append(lastItem+x)
                            continue
                
                        
                    
                
            newStack.append(x)
        
        ans = 0
        print(newStack)
        for x in newStack:
            print(basicDict[x])
            ans += basicDict[x]
        
        return ans    